hudi-client/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -244,6 +245,10 @@ private HoodieCleanMetadata runClean(HoodieTable<?> table, HoodieInstant cleanIn
         inflightInstant = cleanInstant;
       }
 
+      // Update Metadata Table before even finishing clean. This ensures that an async clean operation in the
+      // background does not lead to stale metadata being returned from Metadata Table.
+      HoodieMetadata.update(config, cleanerPlan, cleanInstant.getTimestamp());
```

Comment:
Cleaning/Compaction/writing can all log updates in parallel. Do we have a test around this?

Reply:
I dont have Async tests yet. 

We need to decide where to log update for async operations. If we do it at Async operation completion then we may end up in multi writer scenario.

Reply:
Created HUDI-1319